fibonacci.asm

```diff
@@ -17,15 +16,25 @@
 	.data
 
 fibs:	.word   0 : 9         # create an array variable named "fibs" of 9 word-length elements (4 bytes each)
-size:	.word  9              # create a single integer variable named "size" that indicates the length of the array
+prompttext:	.asciiz "How many fibonacci numbers would you like to see?\n"
+errortext:	.asciiz "Error- please only enter an amount from 1 to 9!\n"
 
 ###############################################
 # .text segment. Assembly instructions go here.
 ###############################################
 	.text
+begin:	      li   $v0, 4	    # syscall 4 = print string
+	      la   $a0, prompttext  # store address of prompt string into a0
+	      syscall		    # do the diddly
+	      li   $v0, 5	    # syscall 5 = read int
+	      syscall		    # v0 will hold result
+	      # check if user input is allowed..
+	      subi $s0, $v0, 10     # subtract input from 10 and store in s0
+	      bgez $s0, invalidsize # if s0 >= 0, then it is not a valid size..
+	      blez $v0, invalidsize # if the input is 0 or less, it is not valid.
+	      
```

Comment:
Very good
Simple, efficient, and effective changes
Prints proper error message and prompts user for input again